src/NuGet.Core/NuGet.ProjectModel/ProjectLockFile/PackagesLockFileUtilities.cs

```diff
@@ -374,6 +383,43 @@ private static bool HasP2PDependencyChanged(IEnumerable<LibraryDependency> newDe
             return false;
         }
 
+        /// <summary>
+        /// All the transitive versions enforced by CPVM and present in the current lock file (packages.lock.json) need to match the versions in the central package management file.
```

Comment:
I'm not sure this will handle new centrally managed versions. 
Take the above example, and try it without C defined, then define C (you can have exact versions for simplicity).

Reply:
In the case above the lock file for project A will be created with:
PackageA
  	"type": "Direct",
        "requested": "[1.0.0, )",
        "resolved": "1.0.0",

PackageB
  	"type": "Direct",
        "requested": "[2.0.0, )",
        "resolved": "2.0.0",

PackageC
  	"type": "CentralTransitive",
        "requested": "[3.0.0, )",
        "resolved": "3.1.0",

If the central file is changed to have 
C 3.0.0 ---> C 4.0.0

the lock will invalidate.

The CentralTransitive will invalidate the lock if the requested version is changing. I updated the comments to make this more clear.

Reply:
I think it's clearer now, thanks for updating the comments.